clients/venice-push-job/src/main/java/com/linkedin/venice/hadoop/mapreduce/datawriter/jobs/DataWriterMRJob.java

```diff
@@ -0,0 +1,394 @@
+      FileInputFormat.setInputPaths(jobConf, new Path(pushJobSetting.inputURI));
+
+      /**
+       * Include all the files in the input directory no matter whether the file is with '.avro' suffix or not
+       * to keep the logic consistent with {@link #checkAvroSchemaConsistency(FileStatus[])}.
+       */
+      jobConf.set(AvroInputFormat.IGNORE_FILES_WITHOUT_EXTENSION_KEY, "false");
+
+      jobConf.set(KEY_FIELD_PROP, pushJobSetting.keyField);
+      jobConf.set(VALUE_FIELD_PROP, pushJobSetting.valueField);
+      if (pushJobSetting.etlValueSchemaTransformation != null) {
+        jobConf.set(ETL_VALUE_SCHEMA_TRANSFORMATION, pushJobSetting.etlValueSchemaTransformation.name());
+      }
+
+      if (pushJobSetting.isAvro) {
+        jobConf.set(SCHEMA_STRING_PROP, pushJobSetting.fileSchemaString);
+        jobConf.set(AvroJob.INPUT_SCHEMA, pushJobSetting.fileSchemaString);
+        if (pushJobSetting.generatePartialUpdateRecordFromInput) {
+          jobConf.setBoolean(GENERATE_PARTIAL_UPDATE_RECORD_FROM_INPUT, true);
+          jobConf.set(UPDATE_SCHEMA_STRING_PROP, pushJobSetting.valueSchemaString);
+        }
+        jobConf.setClass("avro.serialization.data.model", GenericData.class, GenericData.class);
+        jobConf.setInputFormat(AvroInputFormat.class);
+        jobConf.setMapperClass(VeniceAvroMapper.class);
+        jobConf.setBoolean(VSON_PUSH, false);
+      } else {
+        jobConf.setInputFormat(VsonSequenceFileInputFormat.class);
+        jobConf.setMapperClass(VeniceVsonMapper.class);
+        jobConf.setBoolean(VSON_PUSH, true);
+        jobConf.set(FILE_KEY_SCHEMA, pushJobSetting.vsonFileKeySchemaString);
+        jobConf.set(FILE_VALUE_SCHEMA, pushJobSetting.vsonFileValueSchemaString);
+      }
+    }
+  }
+
+  private void setupReducerConf(JobConf jobConf, PushJobSetting pushJobSetting) {
+    if (pushJobSetting.isSourceKafka) {
+      jobConf.setOutputKeyComparatorClass(KafkaInputKeyComparator.class);
+      jobConf.setOutputValueGroupingComparator(KafkaInputValueGroupingComparator.class);
+      jobConf.setCombinerKeyGroupingComparator(KafkaInputValueGroupingComparator.class);
+      jobConf.setPartitionerClass(KafkaInputMRPartitioner.class);
+    } else {
+      jobConf.setPartitionerClass(pushJobSetting.mapReducePartitionerClass);
+    }
+    jobConf.setReduceSpeculativeExecution(pushJobSetting.enableReducerSpeculativeExecution);
+    int partitionCount = pushJobSetting.partitionCount * pushJobSetting.amplificationFactor;
+    jobConf.setInt(PARTITION_COUNT, partitionCount);
+    jobConf.setNumReduceTasks(partitionCount);
+    jobConf.setMapOutputKeyClass(BytesWritable.class);
+    jobConf.setMapOutputValueClass(BytesWritable.class);
+    if (pushJobSetting.isSourceKafka) {
+      jobConf.setReducerClass(VeniceKafkaInputReducer.class);
+    } else {
+      jobConf.setReducerClass(VeniceReducer.class);
+    }
+  }
+
+  // Visible for testing
+  public void setJobClientWrapper(JobClientWrapper jobClientWrapper) {
```

Comment:
Just curious, if it is for unit test only, it can be default modifier, no need to be public.

Reply:
There is some problem with the packaging. This particular function is invoked by `VenicePushJob` class, but the code there is triggered by tests